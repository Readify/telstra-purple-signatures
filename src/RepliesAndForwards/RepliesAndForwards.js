import React from 'react';

import { purple } from '../constants';
import { parseMobile, parseLandLine } from '../util';
const { brandInfo } = purple;

const RepliesAndForwards = props => {
  const {
    name,
    title,
    mobile,
    email,
    sigType,
    twitter,
    supportHotline,
    supportEmail,
    brandName,
    brandLink,
    brandLinkName
  } = props;
  const styleObj = {
    color: 'black',
    fontFamily: 'Arial, sans-serif',
    fontSize: '9pt'
  };
  const titleOptional = title ? <>&nbsp;|&nbsp;{title}</> : null;
  return (
    <div>
      <p style={styleObj}>
        --
        <br />
        <b>
          {name}&nbsp;|&nbsp;{brandName}
          {titleOptional}
        </b>
        <br />
        <b>M</b>&nbsp;
        <a
          href={`tel:${parseMobile(mobile).replace(/\s/g, '')}`}
          dangerouslySetInnerHTML={{
            __html: parseMobile(mobile).replace(/\s/g, '&nbsp;')
          }}
          style={{ color: brandInfo.brandLinkColour }}
        />{' '}
        | <b>E</b>&nbsp;
        <a
          href={`mailto:${email}`}
          style={{ color: brandInfo.brandLinkColour }}
        >
          {email}
        </a>{' '}
        |{' '}
        {twitter ? (
          <span>
            <span>
              <b>T</b>&nbsp;
              <a
                href={`https://twitter.com/${twitter.replace('@', '')}`}
                rel="noopener noreferrer"
                target="_blank"
                style={{ color: brandInfo.brandLinkColour }}
              >
                @{twitter.replace('@', '')}
              </a>
            </span>{' '}
            |{' '}
          </span>
        ) : null}
        <b>W</b>&nbsp;
        <a href={brandLink} style={{ color: brandInfo.brandLinkColour }}>
          {brandLinkName}
        </a>
        {sigType === 'support' ? (
          <span>
            <br />
            <b>Support&nbsp;Hotline</b>&nbsp;
            <a
              href={`tel:${parseLandLine(supportHotline).replace(
                /&nbsp;/g,
                ''
              )}`}
              dangerouslySetInnerHTML={{
                __html: parseLandLine(supportHotline)
              }}
              style={{ color: brandInfo.brandLinkColour }}
            />{' '}
            | <b>Support&nbsp;Email</b>&nbsp;
            <a
              href={`mailto:${supportEmail}`}
              style={{ color: brandInfo.brandLinkColour }}
            >
              {supportEmail}
            </a>
          </span>
        ) : null}
      </p>
      <div
        style={{
          color: '#4a4a4a',
          fontFamily: 'Arial, sans-serif',
          fontSize: '6.5pt',
          lineHeight: '1.5'
        }}
      >
        This email may contain confidential information.
        <br />
        If you are not the intended recipient, please delete it immediately.
      </div>
    </div>
  );
};

export default RepliesAndForwards;
