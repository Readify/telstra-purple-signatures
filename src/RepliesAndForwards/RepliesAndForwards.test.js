import React from 'react';
import { shallow } from 'enzyme';

import RepliesAndForwards from './RepliesAndForwards';

it('Renders', () => {
  const props = {
    email: 'your.name@purple.telstra.com',
    sigType: false,
    mobile: '+61 111 222 333',
    name: 'your name',
    pronoun: 'he/him | she/her | they/them',
    qualifications: 'Jedi Master | PSM I',
    title: 'Job Title',
    twitter: '@myTwitter',
    brandName: 'Telstra Purple',
    brandLink: 'https://purple.telstra.com/',
    brandLinkName: 'purple.telstra.com',
    brandLogo: {
      link:
        'https://readifysignatures.blob.core.windows.net/images/purple-email-logo.png',
      alt: 'Telstra Purple'
    }
  };

  const tree = shallow(<RepliesAndForwards {...props} />);
  expect(tree).toMatchSnapshot();
});
