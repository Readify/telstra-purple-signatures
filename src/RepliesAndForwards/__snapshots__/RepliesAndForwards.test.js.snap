// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders 1`] = `
<div>
  <p
    style={
      Object {
        "color": "black",
        "fontFamily": "Arial, sans-serif",
        "fontSize": "9pt",
      }
    }
  >
    --
    <br />
    <b>
      your name
       | 
      Telstra Purple
       | 
      Job Title
    </b>
    <br />
    <b>
      M
    </b>
     
    <a
      dangerouslySetInnerHTML={
        Object {
          "__html": "+61&nbsp;111&nbsp;222&nbsp;333",
        }
      }
      href="tel:+61111222333"
      style={
        Object {
          "color": "#0E39FB",
        }
      }
    />
     
    | 
    <b>
      E
    </b>
     
    <a
      href="mailto:your.name@purple.telstra.com"
      style={
        Object {
          "color": "#0E39FB",
        }
      }
    >
      your.name@purple.telstra.com
    </a>
     
    |
     
    <span>
      <span>
        <b>
          T
        </b>
         
        <a
          href="https://twitter.com/myTwitter"
          rel="noopener noreferrer"
          style={
            Object {
              "color": "#0E39FB",
            }
          }
          target="_blank"
        >
          @
          myTwitter
        </a>
      </span>
       
      |
       
    </span>
    <b>
      W
    </b>
     
    <a
      href="https://purple.telstra.com/"
      style={
        Object {
          "color": "#0E39FB",
        }
      }
    >
      purple.telstra.com
    </a>
  </p>
  <div
    style={
      Object {
        "color": "#4a4a4a",
        "fontFamily": "Arial, sans-serif",
        "fontSize": "6.5pt",
        "lineHeight": "1.5",
      }
    }
  >
    This email may contain confidential information.
    <br />
    If you are not the intended recipient, please delete it immediately.
  </div>
</div>
`;
